### PR TITLE
Update channels and chaincodes values structure

### PR DESCRIPTION
**Pull request by Kelvin-M**
For HLF 2.x compatibility

Companion PR : 
https://github.com/SubstraFoundation/substra-chaincode/pull/127
https://github.com/SubstraFoundation/hlf-k8s/pull/97

### CODE REVIEW

File: backend/substrapp/ledger/connection.py [hunk 1]
```diff
@@ -113,7 +116,7 @@ def _get_hfc(channel_name):
             peer,
             config=True,
             local=False,
-            interests=[{'chaincodes': [{'name': settings.LEDGER_CHAINCODE_NAME}]}]
+            interests=[{'chaincodes': [{'name': chaincode_name}]}]
```
Comment (Kelvin-M) [inline backend/substrapp/ledger/connection.py:119]: `_lifecycle`

File: backend/requirements.txt [hunk 2]
```diff
@@ -17,7 +17,6 @@ uwsgi==2.0.18
 zxcvbn==4.4.28
 boto3==1.11.9
 fabric-sdk-py==1.0.0
-# dev-mode
```
[the remark this answers is not in the document]
Comment (Kelvin-M) [reply backend/requirements.txt:20]: My bad, collateral removal

File: docker/substra-backend/Dockerfile [hunk 3]
```diff
@@ -1,6 +1,6 @@
 FROM python:3.6-slim AS build
 
-RUN apt-get update && apt-get install gcc -y && apt-get clean
+RUN apt-get update && apt-get install gcc git -y && apt-get clean
```
[the remark this answers is not in the document]
Comment (Kelvin-M) [reply docker/substra-backend/Dockerfile]: Only for test on my side : removed !

File: backend/substrapp/ledger/connection.py [hunk 4]
```diff
@@ -86,25 +86,25 @@ def _get_hfc(channel_name):
...
-    chaincodes = [cc.name
-                  for resp in responses
-                  for cc in resp.chaincodes]
-
-    if settings.LEDGER_CHAINCODE_NAME not in chaincodes:
-        raise Exception(f'Chaincode : {settings.LEDGER_CHAINCODE_NAME}'
-                        f' is not instantiated in the channel :  {channel_name}')
+    # chaincode_name = settings.LEDGER_CHANNELS[channel_name]['chaincode']['name']
+
+    # /!\ New chaincode lifecycle.
```
[the remark this answers is not in the document]
Comment (Kelvin-M) [reply backend/substrapp/ledger/connection.py:95]: Yes !

File: charts/substra-backend/CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,10 @@
 # Changelog
 
+
+## 1.6.0
+
+- Refactor channels / chaincodes values
```
[the remark this answers is not in the document]
Comment (Kelvin-M) [reply charts/substra-backend/CHANGELOG.md]: Done

**Review by AurelienGasser — APPROVED** [review]
Nice work!

**Review by inalgnu — APPROVED** [review]

**Pull request merged**